action/protocol/staking/candidate_statemanager.go

```diff
@@ -178,6 +179,12 @@ func (csm *candSM) Commit() error {
 	return csm.WriteView(_protocolID, csm.DirtyView())
 }
 
+func (csm *candSM) SwitchCandCenter(cc *CandidateCenter) {
+	if cc != nil {
```

Comment:
in the previous implementation, changes are saved in `workingset`, via `dock`.

Reply:
fixed, use `ResetBase()` in latest commit